src/MongoDB.Driver.Core/Core/Configuration/CompressorConfiguration.cs

```diff
@@ -23,24 +24,90 @@ namespace MongoDB.Driver.Core.Configuration
     /// </summary>
     public sealed class CompressorConfiguration
     {
+        // private fields
+        private readonly IDictionary<string, object> _properties;
+        private readonly CompressorType _type;
+
+        // constructors
         /// <summary>
         /// Initializes an instance of <see cref="CompressorConfiguration"/>.
         /// </summary>
         /// <param name="type">The compressor type.</param>
         public CompressorConfiguration(CompressorType type)
         {
-            Type = type;
-            Properties = new Dictionary<string, object>();
+            _type = type;
+            _properties = new Dictionary<string, object>();
         }
 
+        // public properties
         /// <summary>
         /// Gets the compression properties.
         /// </summary>
-        public IDictionary<string, object> Properties { get; }
+        public IDictionary<string, object> Properties => _properties;
```

Comment:
It's odd that you can't set the `Properties` directly, but that you have to modify the *contents* of `Properties`.

But I'm not changing the public API here.

Reply:
if I'm not mistaken, the below code allows setting `Properties` directly without changing `public API`

```
public IDictionary<string, object> Properties { get; private set; }
```

Reply:
Maybe I don't see something, but which line you set `_properties` field in this class (except `constructor`)?

Reply:
I did not change the public API.

I simply switched to an explicit backing field.

Correct, `_properties` is only set in the constructor. Same as before.

Reply:
ok, probably I misunderstood you comment

Reply:
Probably. I meant that the application can't write code like this:

    var properties = new Dictionary<string, object> { { "Level", 1 };
    var configuration = new CompressorConfiguration(CompressorType.ZLib, properties);

and instead must write:

    var configuration = new CompressorConfiguration(CompressorType.ZLib);
    configuration.Properties.Add("Level", 1); // can't pass in properties, have to modify them in place

Reply:
I've got it. Thanks for the clarification.